src/snakefile

```diff
@@ -76,251 +77,322 @@ osemosysFiles = [
+		Path(inputDir, 'EmissionPenaltyByYear.csv'),
+		Path(inputDir, 'USA_Data.xlsx')
 	output:
-		outputDir + 'data/EmissionsPenalty.csv'
-	shell:
-		'python ' + scriptsDir + 'EmissionPenalty.py'
+		Path(outputDir, 'data', 'EmissionsPenalty.csv')
+	log:
+		'logs/emission_penalty.log'
+	conda:
+		'envs/preprocess.yaml'
+	script:
+		f'{scriptsDir}/EmissionPenalty.py'
 
-rule RETags:
+rule re_tags:
 	input:
-		outputDir + 'data/YEAR.csv',
-		outputDir + 'data/REGION.csv',
-		inputDir + 'USA_Data.xlsx'
+		Path(outputDir, 'data', 'YEAR.csv'),
+		Path(outputDir, 'data', 'REGION.csv'),
+		Path(inputDir, 'USA_Data.xlsx')
 	output:
-		outputDir + 'data/RETagTechnology.csv'
-	shell:
-		'python ' + scriptsDir + 'RETags.py'
+		Path(outputDir, 'data', 'RETagTechnology.csv')
+	log:
+		'logs/re_tags.log'
+	conda:
+		'envs/preprocess.yaml'
+	script:
+		f'{scriptsDir}/RETags.py'
 
-rule TechToFromStorage:
+rule tech_to_from_storage:
 	input:
-		outputDir + 'data/YEAR.csv',
-		outputDir + 'data/REGION.csv',
-		inputDir + 'USA_Data.xlsx'
+		Path(outputDir, 'data', 'YEAR.csv'),
+		Path(outputDir, 'data', 'REGION.csv'),
+		Path(inputDir, 'USA_Data.xlsx')
 	output:
-		outputDir + 'data/TechnologyToStorage.csv',
-		outputDir + 'data/TechnologyFromStorage.csv'
-	shell:
-		'python ' + scriptsDir + 'TechToFromStorage.py'
+		Path(outputDir, 'data', 'TechnologyToStorage.csv'),
+		Path(outputDir, 'data', 'TechnologyFromStorage.csv')
+	log:
+		'logs/tech_to_from_storage.log'
+	conda:
+		'envs/preprocess.yaml'
+	script:
+		f'{scriptsDir}/TechToFromStorage.py'
 
-rule TotalAnnualMaxCapacity:
+rule total_annual_max_capacity:
 	input:
-		inputDir + 'USA_Data.xlsx'
+		Path(inputDir, 'USA_Data.xlsx')
 	output:
-		outputDir + 'data/TotalAnnualMaxCapacity.csv'
-	shell:
-		'python ' + scriptsDir + 'TotalAnnualMaxCapacity.py'
+		Path(outputDir, 'data', 'TotalAnnualMaxCapacity.csv')
+	log:
+		'logs/total_annual_max_capacity.log'
+	conda:
+		'envs/preprocess.yaml'
+	script:
+		f'{scriptsDir}/TotalAnnualMaxCapacity.py'
 
-rule otooleConvert:
+rule otoole_convert:
 	input:
-		datapackage = outputDir + 'datapackage.json',
-		csvFiles = expand(outputDir + 'data/{osemosysFile}', 
+		datapackage = Path(outputDir, 'datapackage.json'),
+		csvFiles = expand(Path(outputDir, 'data', '{osemosysFile}'), 
 			osemosysFile = osemosysFiles)
 	output:
-		outputDir + fileName + '.txt'
-	shell:
-		'otoole convert datapackage datafile {input.datapackage} {output}'
-
-rule createLP:
-	input:
-		modelFile = outputDir + 'osemosys_fast_TB.txt',
-		dataFile = outputDir + fileName + '.txt'
-	output:
-		outputDir + fileName + '.lp'
+		f'{fileName}.txt'
+	log:
+		'logs/otoole_convert.log'
+	conda:
+		'envs/otoole_convert.yaml'
 	shell:
-		'glpsol -m {input.modelFile} -d {input.dataFile} --wlp {output} --check'
+		'otoole convert datapackage datafile '
+		'{input.datapackage} {output} 2> {log}'
 
-
-#####################################
-## CLEANING RULES
-#####################################
+#rule create_lp:

```

Comment:
This is a new commented out rule. Do you intend to uncomment it later?

Reply:
This is the same as `rule createLp` from the old file I think. Same as previous though, as it will be addressed with a future issue. You think we should delete it now tho?